src/main/java/org/opensearch/security/resources/ResourceSharingIndexHandler.java

```diff
@@ -567,6 +573,151 @@ public void revoke(String resourceId, String resourceIndex, ShareWith revokeAcce
         }, listener::onFailure);
     }
 
+    /**
+     * Helper method to build shareWith object from the supplied patch to then be used to apply the patch
+     * @param patch source content
+     * @return the parsed content map of entities to share and to revoke
+     */
+    @SuppressWarnings("unchecked")
```

Comment:
IMO I think we should do XContent parsing like is done with ResourceSharing here: https://github.com/opensearch-project/security/blob/main/spi/src/main/java/org/opensearch/security/spi/resources/sharing/ResourceSharing.java#L165-L198

Reply:
done. I've refactored request parser completely.